baybe/constraints/conditions.py

```diff
@@ -126,6 +159,13 @@ def _validate_tolerance(self, _: Any, value: float) -> None:  # noqa: DOC101, DO
                     f"or <= 0.0, but was {value}."
                 )
 
+    def generate_operator_function(self):
+        """Generate a function using operators to filter out undesired rows."""
+        func = rpartial(_threshold_operators[self.operator], self.threshold)
+        if self.operator in _valid_tolerance_operators:
+            func = rpartial(func, atol=self.tolerance)
```

Comment:
I'm not sure how things are being executed internally, but could this three-layer nesting of partials that is involved kill performance?

Reply:
hmm thats something I would really not like to investigate in this PR. Shouldnt this be known in general for Python?

Reply:
Yeah, it's fine here. Just wanted to raise awareness. In the end, it depends very much on how the function is being called, I guess. If a single call is applied to an entire array via broadcasting, then no worries at all (that's how it would happen in numpy). However, I'm just not familiar enough with polars to speculate what happens there. In the worst case, the call is repeated for every single row, and that could really cause some major overhead. But ok, let's try to avoid premature optimization here until this really becomes a problem 👍🏼